tests/ports/rp2/rp2_machine_timer.py

```diff
@@ -0,0 +1,20 @@
+from machine import Timer
+from time import sleep_ms
+
+# Test the rp2-specific adjustable tick_hz and hard/soft IRQ handlers
+# for both one-shot and periodic timers.
+
+modes = {Timer.ONE_SHOT: "one-shot", Timer.PERIODIC: "periodic"}
+kinds = {False: "soft", True: "hard"}
+
+for mode in modes:
+    for hard in kinds:
+        for period in 2, 4:
```

Comment:
This is technically not testing period=4 for periodic mode, because it's only executed once in the 5ms waiting window.

I suggest increasing the waiting window to 9ms.

Reply:
Fair point! I'll push a version that does this.